File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/controller/user/WithdrawUserController.java
package dev.starryeye.book_library.controller.user;

import dev.starryeye.book_library.application.command.WithdrawUserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class WithdrawUserController {

    private final WithdrawUserService service;

    @DeleteMapping("/api/v1/users")
    public void withdrawUser(@RequestParam("id") Long id) {
        service.withdraw(id);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/controller/user/WithdrawUserController.kt
package dev.starryeye.book_library.controller.user

import dev.starryeye.book_library.application.user.command.WithdrawUserService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class WithdrawUserController(
    private val service: WithdrawUserService
) {

    @DeleteMapping("/api/v1/users/")
    fun deleteUser(
        @RequestParam("id") id: Long
    ) {
        service.withdraw(id)
    }
}